x-pack/plugins/ecs_data_quality_dashboard/server/lib/fetch_available_indices.ts

```diff
@@ -4,18 +4,70 @@
  * 2.0; you may not use this file except in compliance with the Elastic License
  * 2.0.
  */
+
 import type { ElasticsearchClient } from '@kbn/core/server';
-import { getRequestBody } from '../helpers/get_available_indices';
+import type { CatIndicesIndicesRecord } from '@elastic/elasticsearch/lib/api/types';
+import dateMath from '@kbn/datemath';
+
+export type FetchAvailableCatIndicesResponseRequired = Array<
+  Required<Pick<CatIndicesIndicesRecord, 'index' | 'creation.date'>>
+>;
 
-type AggregateName = 'index';
-interface Result {
-  index: {
-    buckets: Array<{ key: string }>;
-    doc_count: number;
+export interface FetchAvailableIndicesResponse {
+  aggregations: {
+    index: {
+      buckets: Array<{
+        key: string;
+      }>;
+    };
   };
 }
 
-export const fetchAvailableIndices = (
+export const fetchAvailableIndices = async (
   esClient: ElasticsearchClient,
   params: { indexPattern: string; startDate: string; endDate: string }
-) => esClient.search<AggregateName, Result>(getRequestBody(params));
+): Promise<FetchAvailableIndicesResponse> => {
+  const { indexPattern, startDate, endDate } = params;
+
+  const startDateMoment = dateMath.parse(startDate);
+  const endDateMoment = dateMath.parse(endDate, { roundUp: true });
+
+  if (
+    !startDateMoment ||
+    !endDateMoment ||
+    !startDateMoment.isValid() ||
+    !endDateMoment.isValid()
+  ) {
+    throw new Error('Invalid date format in startDate or endDate');
+  }
+
+  const startDateMillis = startDateMoment.valueOf();
+  const endDateMillis = endDateMoment.valueOf();
+
+  const indices = (await esClient.cat.indices({
+    index: indexPattern,
+    format: 'json',
+    h: 'index,creation.date',
+  })) as FetchAvailableCatIndicesResponseRequired;
+
+  const filteredIndices = indices.filter((indexInfo) => {
+    const creationDate: string = indexInfo['creation.date'] ?? '';
+    const creationDateMillis = parseInt(creationDate, 10);
+
+    if (isNaN(creationDateMillis)) {
+      return false;
+    }
+
+    return creationDateMillis >= startDateMillis && creationDateMillis <= endDateMillis;
+  });
+
+  return {
+    aggregations: {
+      index: {
+        buckets: filteredIndices.map((indexInfo) => ({
+          key: indexInfo.index,
+        })),
+      },
+    },
+  };
```

Comment:
I think we can make it simpler and return an array here, as we are no longer doing any aggregation in the query
```
filteredIndices.map((indexInfo) => ({
          key: indexInfo.index,
        }))
```

Reply:
Can we also deploy this PR to the cloud and check it again before merge?

Reply:
Yes also saw that after push and started simplifying. Thanks for catching tho.
I will add the label. Thanks